spec/enumerable_spec.rb

```diff
@@ -0,0 +1,80 @@
+require "./enumerable.rb"
+
+RSpec.describe Enumerable do 
+  let(:array) { [4,5,6] }
+  let(:my_each_arr) { array.my_each { |n| n } } 
+  myindex = Proc.new { [4,5,6].my_each_with_index { |val,index| index } } 
+  mapy = Proc.new { [4,5,6].my_map {|num| num * 10 } }
+  selecty = Proc.new { [4,5,6].my_select { |n| n.even? } }
+
+  describe "#my_each" do 
+    it "it tests my each" do 
+      expect(my_each_arr).to eq(array)
+		end
+  end
+  
+  describe "#my_each_with_index" do 
+    it "it tests my each with index" do
+      expect(array.my_each_with_index { |x, i| array[i] = i }).to eql([0, 1, 2])
+    end
+  end
+
+  describe "#my select" do
+    it "it test my select" do 
+      expect([4,5,6].my_select { |n| n.even? } ).to eq([4,6] )
+    end
+  end
+
+  describe "#my_all?" do 
+    it "it tests my all?" do 
+      expect(array.my_all? { |x| x > 3 }).to eq(true)
+    end
+
+    it "it tests my all?" do 
+      expect(array.my_all? { |x| (Numeric) }).to eq(true)
+    end
+
+    it "it tests my all?" do 
+      expect(array.my_all? { |x| x > 7 } ).to eq(false)
+    end
+  end
+
+  describe "#my_any" do
+    it "it tests if any greater than 7" do
+      expect(array.my_any? { |x| x > 7 } ).to eq(false)
+    end
+
+    it "it tests if any equal to 4" do
```

Comment:
Please do describe a general behavior, this is a bit too explicit and does not describe in detail the behavior being tested.